src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5783,46 +5826,59 @@ int wm_vuldet_fetch_MSU(update_node *update, char *repo) {
     }
 
     os_free(hash);
-
     return 0;
 }
 
-bool wm_vuldet_update_MSU(update_node *update) {
-    feed_metadata *msu = NULL;
-    if (msu = wm_vuldet_fetch_MSU_metadata(update), msu == NULL){
-        return true;
-    }
-
-    bool retval = true;
+bool wm_vuldet_update_wazuh_feed(update_node *update) {
+    feed_metadata   *metadata_feed = NULL;
+    bool            retval = true;
 
-    if (!msu->sha256 || !msu->last_mod) {
+    if (metadata_feed = wm_vuldet_fetch_feed_metadata(update), metadata_feed == NULL){
+        return retval;
+    }
+    
+    if (!metadata_feed->sha256 || !metadata_feed->last_mod) {
         goto clean;
     }
 
     if (wm_vuldet_check_timestamp(vu_feed_tag[update->dist_tag_ref],
-        msu->last_mod, NULL) == VU_TIMESTAMP_UPDATED) {
+        metadata_feed->last_mod, NULL) == VU_TIMESTAMP_UPDATED) {
         mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_UPDATE_DATE, update->dist_ext);
         retval = false;
     } else {
-        wm_insert_MSU_metadata(msu, update);
+        wm_insert_feed_metadata(metadata_feed, update);
     }
 
 clean:
-    os_free(msu->version);
-    os_free(msu->sha256);
-    os_free(msu->last_mod);
-    os_free(msu->size);
-    os_free(msu->g_size);
-    os_free(msu);
+    os_free(metadata_feed->version);
+    os_free(metadata_feed->sha256);
+    os_free(metadata_feed->last_mod);
+    os_free(metadata_feed->size);
+    os_free(metadata_feed->g_size);
+    os_free(metadata_feed);
 
     return retval;
 }
 
-feed_metadata * wm_vuldet_fetch_MSU_metadata(update_node *update) {
-    int res_url_request;
+feed_metadata * wm_vuldet_fetch_feed_metadata(update_node *update) {
+    int     res_url_request;
+    char    *metadata_repo;
+    
+    // An additional metadata feed is provided for MSU and ALAS feeds
+    switch (update->dist_ref) {
+        case FEED_MSU:
+            metadata_repo = MSU_REPO_META;
+            break;
+        case FEED_ALAS:
+            metadata_repo = ALAS_REPO_META;
+            break;
+        default:
+            merror("No metadata feed available for (%s)", vu_feed_ext[update->dist_ref]);
```

Comment:
This error, when evaluating Amazon Linux, won't specify for what version we are fetching the metadata file